File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/widget/textview/LabelTextView.java
package com.didichuxing.doraemonkit.widget.textview;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.AttrRes;
import androidx.annotation.ColorRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import com.didichuxing.doraemonkit.R;

/**
 * Created by jinliang on 2017/9/29.
 */

public class LabelTextView extends LinearLayout {
    private TextView mLabel;
    private TextView mText;

    public LabelTextView(@NonNull Context context) {
        this(context, null);
    }

    public LabelTextView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public LabelTextView(@NonNull Context context, @Nullable AttributeSet attrs, @AttrRes int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        setOrientation(HORIZONTAL);
        LayoutInflater.from(context).inflate(R.layout.dk_label_text_view, this);

        TypedArray a = getContext().obtainStyledAttributes(attrs, R.styleable.LabelTextView);
        String label = a.getString(R.styleable.LabelTextView_dkLabel);
        String text = a.getString(R.styleable.LabelTextView_dkText);
        int maxLines = a.getInt(R.styleable.LabelTextView_dkMaxLines, 0);
        a.recycle();

        mLabel = findViewById(R.id.label);
        mText = findViewById(R.id.text);

        setLabel(label);
        setText(text);
        if (maxLines > 0) {
            mText.setMaxLines(maxLines);
        }
    }

    public void setText(String text) {
        mText.setText(text);
    }

    public void setText(@StringRes int text) {
        mText.setText(text);
    }

    public void setLabel(String label) {
        mLabel.setText(label);
    }

    public void setLabel(@StringRes int label) {
        mLabel.setText(label);
    }

    public void setTextColor(@ColorRes int color) {
        mText.setTextColor(getResources().getColor(color));
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/widget/textview/LabelTextView.kt
package com.didichuxing.doraemonkit.widget.textview

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import android.widget.TextView
import androidx.annotation.ColorRes
import androidx.annotation.StringRes
import androidx.core.content.ContextCompat
import com.didichuxing.doraemonkit.R

/**
 * @author lostjobs created on 2020/6/9
 */
class LabelTextView : LinearLayout {

  private lateinit var mLabel: TextView
  private lateinit var mText: TextView

  constructor(context: Context) : this(context, null)
  constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
  constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
    context,
    attrs,
    defStyleAttr
  ) {
    initialize(context, attrs, defStyleAttr)
  }

  private fun initialize(context: Context, attrs: AttributeSet?, defStyleAttr: Int) {
    orientation = HORIZONTAL
    LayoutInflater.from(context).inflate(R.layout.dk_label_text_view, this)
    mLabel = findViewById(R.id.label)
    mText = findViewById(R.id.text)
    context.theme.obtainStyledAttributes(attrs, R.styleable.LabelTextView, defStyleAttr, 0)
      .apply {

        mLabel.text = getString(R.styleable.LabelTextView_dkLabel)
        mText.text = getString(R.styleable.LabelTextView_dkText)
        val maxLines = getInt(R.styleable.LabelTextView_dkMaxLines, 0)
        if (maxLines > 0) {
          mText.maxLines = maxLines
        }
        recycle()
      }
  }

  fun setText(text: String?) {
    mText.text = text
  }

  fun setText(@StringRes text: Int) {
    mText.setText(text)
  }

  fun setLabel(label: String?) {
    mLabel.text = label
  }

  fun setLabel(@StringRes label: Int) {
    mLabel.setText(label)
  }

  fun setTextColor(@ColorRes color: Int) {
    mText.setTextColor(ContextCompat.getColor(context, color))
  }

}